NDTensors/src/lib/TypeParameterAccessors/test/runtests.jl

```diff
@@ -0,0 +1,180 @@
+    ## When it is called so I believe its actually working properly
+    ## In a wrapped function on the command line this does however show the correct
+    ## value when using code_warntype.
+    @test ((() -> set_ndims(Array{<:Any,3}, 4))()) == Array{<:Any,4}
+    @test ((() -> set_eltype(Array{<:Any,3}, Float16))()) == Array{Float16,3}
+  end
+
+  @testset "Set a parameter if it is unspecified" begin
+    @test @inferred(specify_parameter(Array{Float32,3}, 1, Float16)) == Array{Float32,3}
+    @test @inferred(specify_parameter(Array{Float32,3}, 2, 2)) == Array{Float32,3}
+    @test @inferred((() -> specify_parameter(Array{Float32}, 1, Float16))()) ==
+      Array{Float32}
+    @test @inferred((() -> specify_parameter(Array{Float32}, 2, 2))()) == Array{Float32,2}
+    @test @inferred((() -> specify_parameter(Array, 1, Float16))()) == Array{Float16}
+    @test @inferred((() -> specify_parameter(Array, 2, 2))()) == Array{<:Any,2}
+    @test @inferred((() -> specify_parameter(Matrix, 1, Float32))()) == Array{Float32,2}
+    @test @inferred((() -> specify_parameter(Matrix, 2, 2))()) == Array{<:Any,2}
+    @test @inferred((() -> specify_parameter(Vector, 1, Float64))()) == Array{Float64,1}
+    @test @inferred((() -> specify_parameter(Vector, 2, 2))()) == Array{<:Any,1}
+
+    @test specify_parameter(Array{<:Any,3}, 1, Float16) == Array{Float16,3}
+    @test specify_parameter(Array{<:Any,3}, 2, 2) == Array{<:Any,3}
+    ## TODO Do we want to support type stable this behavior, it does not seem suppported by julia.
+    ## Though if you have `Matrix` or `Vector` the code is type stable
+    @test_broken @inferred(specify_parameter(Array{<:Any,3}, 1, Float16)) ==
+      Array{Float16,3}
+    @test_broken @inferred(specify_parameters(Array{<:Any,3}, 2, 2)) == Array{<:Any,3}
+  end
+
+  @testset "Set multiple parameters if they are unspecified" begin
+    @test @inferred(
+      (() -> specify_parameter(specify_parameter(Array, 1, Float32), 2, 2))()
+    ) == Array{Float32,2}
+    @test @inferred(
+      (() -> specify_parameter(specify_parameter(Array, 2, 2), 1, Float32))()
+    ) == Array{Float32,2}
+
+    @test @inferred(
+      (() -> specify_parameter(specify_parameter(Matrix, 2, 3), 1, Float32))()
+    ) == Array{Float32,2}
+    @test @inferred(
+      (() -> specify_parameter(specify_parameter(Matrix, 1, Float32), 2, 3))()
+    ) == Array{Float32,2}
+    @test @inferred(
+      (() -> specify_parameter(specify_parameter(Vector, 1, Float32), 2, 3))()
+    ) == Array{Float32,1}
+  end
+
+  ## TODO Still working on the Default parameters system
+  @testset "Default parameters" begin
+    @test @inferred(default_parameter(Array{Float32,3}, Position(1))) == Float64
+    @test @inferred(default_parameter(Array{Float32,3}, Position(2))) == 1
+  end
+
+  @testset "Set to the default parameter" begin
+    @test @inferred(set_parameters(Array{Float32,3}, Position(1), DefaultParameter())) ==
```

Comment:
I think we only want to use `set_parameter` for these kinds of calls, not `set_parameters`. `set_parameters` for now could just be setting all parameters like `set_parameters(Array{Float32,3}, (Float64, 2)) == Matrix{Float64}`, or removed entirely (maybe it is still needed internally for `specify_parameters` but we could remove it from the external interface of the package).

Reply:
Okay that makes sense. I still haven't worked on the `DefaultParameter` system so I will look into this today.

Reply:
@kmp5VT is this resolved? I'm trying to close resolved or outdated comments to track what we still need to change.